DatadogInternal/Sources/BacktraceReporting/BacktraceReporter.swift

```diff
@@ -6,13 +6,41 @@
 
 import Foundation
 
+/// A type representing unique thread identifier for `BacktraceReport` generation.
+public struct ThreadID {
+    public let thread_t: thread_t
+}
+
 /// A protocol for types capable of generating backtrace reports.
 public protocol BacktraceReporting {
-    /// Generates a backtrace report.
-    /// - Returns: A `BacktraceReport` containing information about the current state of all running threads in the process,
-    ///            focusing on tracing back from the error point to the root cause or the origin of the problem. Returns `nil` if
-    ///            the backtrace report cannot be generated.
-    func generateBacktrace() -> BacktraceReport?
+    /// Generates a backtrace report for given thread ID.
+    ///
+    /// The thread given by `threadID` will be promoted in the main stack of returned `BacktraceReport` (`report.stack`).
+    ///
+    /// - Parameter threadID: An ID of the thread that backtrace generation should start on.
+    /// - Returns: A `BacktraceReport` starting on the given thread and containing information about all other threads
+    ///            running in the process. Returns `nil` if the backtrace report cannot be generated.
+    func generateBacktrace(threadID: ThreadID) -> BacktraceReport?
+}
+
+public extension BacktraceReporting {
+    /// Obtains the `ThreadID` of the caller thread. 
+    /// 
+    /// Should be used in conjunction with `generateBacktrace(threadID:)` to generate backtrace of particular thread.
+    func currentThreadID() -> ThreadID {
```

Comment:
looks more suited as extension on `ThreadID` as static method.

Reply:
Fair call 👍, changed to:
```swift
/// A value identifying the thread for `BacktraceReport` generation.
public typealias ThreadID = thread_t

public extension Thread {
    /// Obtains the `ThreadID` of the caller thread.
    ///
    /// Should be used in conjunction with `BacktraceReporting.generateBacktrace(threadID:)` to generate backtrace of particular thread.
    static var currentThreadID: ThreadID { pthread_mach_thread_np(pthread_self()) }
}
```